Cattac/Cattac/GameEngine.swift

```diff
@@ -26,14 +26,17 @@ class GameEngine {
 
     // The AI engine that is used when multiplayer mode is not active
     private var gameAI: GameAI!
+    
+    /// States to advance, initialized at 1 to rollover PreStart state.
+    private var statesToAdvance: Int = 1
```

Comment:
`statesToAdvance == 2` after performActions call when no actions (hence the state advance during `nextState` problem). Game would be stuck in ActionsExecution